paraatm/bdn/_bdn.py

```diff
@@ -0,0 +1,157 @@
+"""
+Bayesian deep neural network (BDN) class
+---
+Required pakages:
+keras
+numpy
+
+"""
+import keras.backend as K
+from keras.models import Model
+from keras.layers import Input, Dense, Dropout,LSTM,Activation
+from keras.optimizers import Adam
+from keras.regularizers import l1,l2
+from keras.models import Sequential
+import numpy as np
+
+class BDN(object):
+    """
+    BDN class creates bayesian deep neural network and provides 
+    predictions with uncertainty
+    
+    Reference: Gal, Y., & Ghahramani, Z. (2016, June). Dropout as a bayesian approximation: Representing model uncertainty in deep learning. In international conference on machine learning (pp. 1050-1059).
+    
+    Attributes
+    ----------
+    x_train: training data
+    y_train: training target
+    x_test: test data
+    y_test: test target
+    idrop: dropout rate for input layer
+    odrop: dropout rate for output layer
+    rdrop:dropout rate for the recurrent layer(DNN model doesn't need rdrop
+    but is provided by default)
+    weight_decay: regularization factor
+    lr: learning rate
+    num_unit: number of unit at each layer
+    batch_size: mini batch size
+    epochs: number of epochs
+    iter_: number of predictions for each sample
+    results: array of predicted results for the test data.
+    
+    Methods:
+    -------
+    get_RNN_model
+    get_DNN_model
+    BDNpred
+    
+    """
+    def __init__(self,x_train,y_train,x_test,y_test,idrop=0.,
+                 odrop=0.25,rdrop=0.25,
+                 weight_decay=1e-4,lr=1e-3,num_unit=100,
+                 batch_size=30,epochs=200,iter_=5):
```

Comment:
Please document this function and its parameters using a docstring and a `Parameters` heading.  See para-atm for examples of format.